pkg/detectors/rabbitmq/rabbitmq.go

```diff
@@ -51,37 +59,49 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 			continue
 		}
 
-		redact := strings.TrimSpace(strings.Replace(parsedURL.String(), password, "********", -1))
-
-		s := detectors.Result{
+		r := detectors.Result{
 			DetectorType: detectorspb.DetectorType_RabbitMQ,
 			Raw:          []byte(urlMatch),
-			Redacted:     redact,
+			Redacted:     strings.TrimSpace(strings.Replace(parsedURL.String(), password, "********", -1)),
 		}
 
 		if verify {
-			conn, err := amqp.Dial(urlMatch)
-			if err == nil {
-				s.Verified = true
-			}
-			if conn != nil {
-				conn.Close()
+			isVerified, verificationErr := s.verify(urlMatch)
+			r.Verified = isVerified
+			if verificationErr != nil {
+				r.SetVerificationError(verificationErr, urlMatch)
 			}
 		}
 
-		if !s.Verified {
+		if !r.Verified {
 			// Skip unverified findings where the password starts with a `$` - it's almost certainly a variable.
 			if strings.HasPrefix(password, "$") {
 				continue
 			}
 		}
 
-		results = append(results, s)
+		results = append(results, r)
 	}
 
 	return results, nil
 }
 
+func (s Scanner) verify(url string) (bool, error) {
+	// Add a timeout.
+	// https://github.com/rabbitmq/amqp091-go/blob/dc67c21576c230f589636319f05b7262915313e6/examples_test.go#L22
+	conn, err := amqp.DialConfig(url, amqp.Config{
+		Dial: func(network, addr string) (net.Conn, error) {
+			return net.DialTimeout(network, addr, 10*time.Second)
+		},
+	})
+	defer func() {
+		if conn != nil {
+			_ = conn.Close()
+		}
+	}()
+	return err == nil, err
```

Comment:
Future work would be to categorize the errors provided by this, differentiating between valid and invalid.

Such as:
`Verification issue: lookup xcws0001: no such host`
`Verification issue: i/o timeout`